agent/src/beerocks/fronthaul_manager/monitor/monitor_thread.cpp

```diff
@@ -213,24 +215,19 @@ bool monitor_thread::init()
         return false;
     }
 
-    auto request =
-        message_com::create_vs_message<beerocks_message::cACTION_MONITOR_JOINED_NOTIFICATION>(
-            cmdu_tx);
-
-    if (request == nullptr) {
-        LOG(ERROR) << "Failed building message!";
-        return false;
-    }
-
-    message_com::send_cmdu(slave_socket, cmdu_tx);
-
     add_socket(slave_socket);
     LOG(DEBUG) << "init() end";
     return true;
 }
 
 void monitor_thread::after_select(bool timeout)
 {
+    if (!m_logger_configured) {
```

Comment:
Please move this to `before_select`

Reply:
Why? Currently, we don't override "before_select" on this thread.

Reply:
> Why?

Because it makes sure that any logs that are done during the first select are also done correctly.

Ideally, it should be done in a thread initialization function, but beerocks_thread_base doesn't have that (init() is called from the context of the parent, not of the thread itself).

> Currently, we don't override "before_select" on this thread.

That's not a good reason :-)

Reply:
Ok added "before_select" and move it to there.